src/pid.cpp

```diff
@@ -126,6 +126,9 @@ void Pid::setGains(double p, double i, double d, double i_max, double i_min, boo
 
 void Pid::setGains(const Gains & gains)
 {
+  if (gains.i_min_ > gains.i_max_) {
+    throw std::invalid_argument("received i_min > i_max");
```

Comment:
IMHO, it would be better to add ERROR here and disable PID then throw exception. What if we have running system and start new controller with bad parameter – then the whole drive crashes. If we disable movement of something would be better, wouldn't be?

Or should we catch this in controller manager, to prevent HW drivers and other controllers to crash?

Reply:
with ERROR, you mean an output on `std::cerr`?

"Disable movement" is not straight forward, isn't it? It could be in some configuration where zero torque would cause the robot to crash -> we cannot solve all failure cases from this generic controller.

I think for the initial setup of the controllers it is safe to "just" throw an exception (nothing is moving, we are in a safe configuration)? In the case of online reconfiguration, we also could just drop the new gains, give an error message, and work with the previous ones.

Reply:
I changed it now, so that the exception is only thrown if the base class is constructed with bad parameters. Otherwise, the new gains are skipped and nothing more happens.